packages/react-form/src/tests/utilities.test.ts

```diff
@@ -97,32 +97,121 @@ describe('getValues()', () => {
+  const formValue = {
+    username: 'myUsername',
+    occupation: 'software developer',
+    name: {
+      firstName: 'myFirstName',
+      lastName: 'myLastName',
+      nickName: {
+        preferred: 'nick name pref',
+        optional: 'nick name optional',
+      },
+    },
+    emails: [
+      {
+        work: 'work@example.com',
+        work2: 'work2@example.com',
+      },
+      {
+        personal: 'personal@example.com',
+      },
+      {
+        hobby: 'hobby@example.com',
+      },
+    ],
+  };
+
   it('returns dirty form values', () => {
-    const formValue = {
-      username: 'myUsername',
+    const fieldBag = {
+      username: {...mockField(formValue.username), dirty: true},
       name: {
-        firstName: 'myFirstName',
+        firstName: mockField(formValue.name.firstName),
       },
       emails: [
         {
-          work: 'work@example.com',
+          work: mockField(formValue.emails[0].work),
+        },
+      ],
+    };
+
+    expect(getDirtyValues(fieldBag)).toStrictEqual({
+      username: formValue.username,
+    });
+  });
+
+  it('returns nested dirty form values up to two nested fields', () => {
+    const fieldBag = {
+      username: mockField(formValue.username),
+      occupation: {...mockField(formValue.occupation), dirty: true},
+      name: {
+        firstName: {...mockField(formValue.name.firstName), dirty: true},
+        lastName: mockField(formValue.name.lastName),
+      },
+      emails: [
+        {
+          personal: {...mockField(formValue.emails[1].personal), dirty: true},
         },
       ],
     };
+
+    expect(getDirtyValues(fieldBag)).toStrictEqual({
+      occupation: formValue.occupation,
+      name: {
+        firstName: formValue.name.firstName,
+      },
+      emails: [
+        {
+          personal: formValue.emails[1].personal,
+        },
+      ],
+    });
+  });
+
+  it('returns deeply nested dirty form values (further nested than library supports)', () => {
     const fieldBag = {
       username: {...mockField(formValue.username), dirty: true},
       name: {
-        firstName: mockField(formValue.name.firstName),
+        firstName: {...mockField(formValue.name.firstName), dirty: true},
+        lastName: mockField(formValue.name.lastName),
+        nickName: {
+          preferred: mockField(formValue.name.nickName.preferred),
+          optional: {
+            ...mockField(formValue.name.nickName.optional),
+            dirty: true,
+          },
+        },
       },
       emails: [
         {
           work: mockField(formValue.emails[0].work),
+          work2: {...mockField(formValue.emails[0].work2), dirty: true},
+        },
+        {
+          personal: {...mockField(formValue.emails[1].personal), dirty: true},
+        },
+        {
+          hobby: mockField(formValue.emails[2].hobby),
         },
       ],
     };
 
-    expect(getDirtyValues(fieldBag)).toStrictEqual({
+    expect(getDirtyValues(fieldBag as any)).toStrictEqual({
```

Comment:
Is there a reason for this `as any` cast here?

Reply:
The reason I casted this to any is to really test the nested functionality. Currently, the library only supports one deep nested fields (top level fields or an object containing other fields), so without this our types will complain 

<img width="1183" alt="image" src="https://user-images.githubusercontent.com/29992628/202562466-53bc5cb9-6d36-4dc5-bcc8-0b7f71ef727b.png">

Given the library only supports this one deep nesting, I was debating to whether even go the recursive approach since we could just brute force check a layer deep, but for JS usage and if this package ever gets updated to support deeper nested fields I went with this approach. Also open to ideas on the implementation given that context ^